apps/search/src/components/ImportProductsToAlgolia/useSearchProvider.ts

```diff
@@ -0,0 +1,28 @@
+import { inferProcedureOutput } from "@trpc/server";
+import { useMemo } from "react";
+
+import { AlgoliaSearchProvider } from "../../lib/algolia/algoliaSearchProvider";
+import { AppRouter } from "../../modules/trpc/trpc-app-router";
+
+type AlgoliaConfiguration = inferProcedureOutput<AppRouter["configuration"]["getConfig"]>;
+
+export const useSearchProvider = (algoliaConfiguration?: AlgoliaConfiguration) => {
+  const searchProvider = useMemo(() => {
+    if (!algoliaConfiguration?.appConfig?.appId || !algoliaConfiguration?.appConfig?.secretKey) {
+      return null;
+    }
+    return new AlgoliaSearchProvider({
+      appId: algoliaConfiguration.appConfig.appId,
+      apiKey: algoliaConfiguration.appConfig.secretKey,
+      indexNamePrefix: algoliaConfiguration.appConfig.indexNamePrefix,
+      enabledKeys: algoliaConfiguration.fieldsMapping.enabledAlgoliaFields,
```

Comment:
you sure you want to execute this on frontend? You can call tRPC here and run algolia server side, so credentials are not exposed to the client

Reply:
If you do this on the frontend, you should verify if app has proper permissions. We should ensure `MANAGE_APPS` is required to configure the app. If we agree on fetching credentials to the frontend, we should ensure credentials will not be retrieved by a staff user without MANAGE_APPS

Reply:
@lkostrowski i didn't touch that logic, it was only moved to the other file, it's even yours: https://github.com/saleor/apps/blob/main/apps/search/src/components/ImportProductsToAlgolia.tsx#L26

Reply:
Fine. So maybe you should look into this later. When the app was created, it was not possible to open apps without MANAGE_APPS permission. Now, app can be accessed so probably we should verify that